lib/widget/flat_button_text.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:survey_flutter_ic/extension/context_extension.dart';
+import 'package:survey_flutter_ic/theme/colors.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+
+class FlatButtonText extends StatefulWidget {
+  final String text;
+  final bool isEnabled;
+  final Function() onPressed;
+
+  const FlatButtonText({
+    super.key,
+    required this.text,
+    required this.isEnabled,
+    required this.onPressed,
+  });
+
+  @override
+  State<FlatButtonText> createState() => _FlatButtonTextState();
+}
+
+class _FlatButtonTextState extends State<FlatButtonText> {
+  @override
+  Widget build(BuildContext context) {
+    return TextButton(
+      onPressed: widget.isEnabled ? widget.onPressed : null,
+      style: TextButton.styleFrom(
+        backgroundColor: Colors.white,
+        padding: const EdgeInsets.symmetric(vertical: space18),
+        shape: const RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(
+            Radius.circular(borderRadius10),
+          ),
+        ),
+      ),
+      child: Text(
+        context.localization.sign_in_button,
+        style: TextStyle(
+          color: widget.isEnabled ? textColorGray : textColorGrayDisabled,
```

Comment:
Can we set the color directly instead of creating `const` value 🤔 
```suggestion
          color: widget.isEnabled ? ColorName.gray : ColorName.black20,
```

Reply:
@Wadeewee Ideally, the colors from resources (`colors.xml`) are supposed as color palettes. Generate new meaningful references for text color, button background, .etc would be easier in themes, styles management IMO 😄

Reply:
Got it @hoangnguyen92dn. Actually, I'm good for both ways 🚀